### PR TITLE
[ticket/14237] Use $language class for notifications

### PR DESCRIPTION
**Pull request by VSEphpbb**
Includes other minor fixes to the notifications classes I found while working on this.

PHPBB3-14237

### CODE REVIEW

**Comment by Nicofuma** [comment]
Couldn't it be possible to get ride of the language dependency here, use language strings everywhere and translate them in the template using the `lang()` function and/or the maybe incoming `lang` filter?

**Comment by VSEphpbb** [comment]
Doesn't seem like it. It's using the sprintf thing in some places, as well as lang's is_set() method for some lang keys (see report_pm for example). Not to mention add_lang()

**Comment by Nicofuma** [comment]
humf ok :/

File: phpBB/phpbb/notification/manager.php [hunk 1]
```diff
@@ -858,7 +873,8 @@ public function get_method_class($method_name)
 	/**
 	 * Helper to load objects (notification types/methods)
 	 *
-	 * @return method\method_interface|type\type_interface
+	 * @param string $object_name
+	 * @return \phpbb\notification\method\method_interface|\phpbb\notification\type\type_interface
```
Comment (Nicofuma) [inline phpBB/phpbb/notification/manager.php]: `method\method_interface|type\type_interface`

File: phpBB/phpbb/notification/manager.php [hunk 2]
```diff
@@ -848,7 +862,8 @@ public function get_item_type_class($notification_type_name, $data = array())
 	/**
 	 * Helper to get the notifications method class and set it up
 	 *
-	 * @return method\method_interface
+	 * @param string $method_name
+	 * @return \phpbb\notification\method\method_interface
```
Comment (Nicofuma) [inline phpBB/phpbb/notification/manager.php]: `method\method_interface`

File: phpBB/phpbb/notification/manager.php [hunk 3]
```diff
@@ -834,7 +846,9 @@ public function get_default_methods()
 	/**
 	 * Helper to get the notifications item type class and set it up
 	 *
-	 * @return type\type_interface
+	 * @param string $notification_type_name
+	 * @param array  $data
+	 * @return \phpbb\notification\type\type_interface
```
Comment (Nicofuma) [inline phpBB/phpbb/notification/manager.php]: `method\method_interface`

You need to be consistent actually, pick one form and use it everywhere

Comment (VSEphpbb) [inline phpBB/phpbb/notification/manager.php]: I think my IDE caused this

**Pull request merged**